config/env_config.go

```diff
@@ -157,7 +157,7 @@ type EnvConfig struct {
 	// Enables endpoint discovery via environment variables.
 	//
 	//	AWS_ENABLE_ENDPOINT_DISCOVERY=true
-	EnableEndpointDiscovery *bool
+	EnableEndpointDiscovery *EndpointDiscoveryEnableState
```

Comment:
Should this be `EndpointDiscoveryState`? Then the constants are `EndpointDiscoveryAuto`, `EndpointDiscoveryEnabled`, `EndpointDiscoveryDisabled`?

Also what is your thought of making the first value (zero-value) as being `EndpointDiscoveryUnset`? Then we could distinguish the default from `auto`, `enabled`, `disabled` without having to use a pointer value.

Reply:
Ah, I failed to notice we exposed this setting on env_config despite not supported the feature at launch. That's unfortunate. Though this is still a breaking change regardless change from `*bool`.

Reply:
Yeah :/ this was missed before GA, since the parameter did nothing and never could be used we should consider doing a minor version bump to change this from an boolean to enum.

Reply:
I don't have a strong opinion on having `EndpointDiscoveryUnset` or not. I do feel having Unset and Auto as values will confuse customers as they are effectively the same in behavior.  

What do we gain by replacing the pointer value? Or is it an idiomatic change?

Reply:
yes breaking but its non-functional  today so seems fine.